ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java

```diff
@@ -64,46 +65,70 @@ static LogBlobRecordInfo readSingleRecordFromLog(RandomAccessFile randomAccessFi
-          isExpired = isExpired(expiresAtMs, currentTimeInMs);
-          blobProperty = " Blob properties - blobSize  " + props.getBlobSize() + " serviceId " + props.getServiceId()
-              + ", isExpired " + isExpired + " accountId " + props.getAccountId() + " containerId "
-              + props.getContainerId();
-          ByteBuffer metadata = MessageFormatRecord.deserializeUserMetadata(streamlog);
-          usermetadata = " Metadata - size " + metadata.capacity();
-          BlobData blobData = MessageFormatRecord.deserializeBlob(streamlog);
-          blobDataOutput = "Blob - size " + blobData.getSize();
+      } else if (version == MessageFormatRecord.Message_Header_Version_V2) {
+        ByteBuffer buffer = ByteBuffer.allocate(MessageFormatRecord.MessageHeader_Format_V2.getHeaderSize());
+        buffer.putShort(version);
+        randomAccessFile.read(buffer.array(), 2, buffer.capacity() - 2);
+        buffer.clear();
+        header = new MessageFormatRecord.MessageHeader_Format_V2(buffer);
+        messageheader =
+            " Header - version " + header.getVersion() + " messagesize " + header.getMessageSize() + " currentOffset "
+                + currentOffset + " blobEncryptionKeyRelativeOffset "
+                + header.getBlobEncryptionKeyRecordRelativeOffset() + " blobPropertiesRelativeOffset "
+                + header.getBlobPropertiesRecordRelativeOffset() + " userMetadataRelativeOffset "
+                + header.getUserMetadataRecordRelativeOffset() + " dataRelativeOffset "
+                + header.getBlobRecordRelativeOffset() + " crc " + header.getCrc();
+        totalRecordSize += header.getMessageSize() + buffer.capacity();
+      } else {
+        throw new MessageFormatException("Header version not supported " + version, MessageFormatErrorCodes.IO_Error);
+      }
+      // read blob id
+      InputStream streamlog = Channels.newInputStream(randomAccessFile.getChannel());
+      blobId = new BlobId(new DataInputStream(streamlog), clusterMap);
+      totalRecordSize += blobId.sizeInBytes();
+      if (header.getBlobPropertiesRecordRelativeOffset()
+          != MessageFormatRecord.Message_Header_Invalid_Relative_Offset) {
+        ByteBuffer blobEncryptionKey = null;
+        if (header.hasEncryptionKeyRecord()) {
+          blobEncryptionKey = MessageFormatRecord.deserializeBlobEncryptionKey(streamlog);
+          encryptionKey = "EncryptionKey found which is of size " + blobEncryptionKey.remaining();
         } else {
-          DeleteRecord deleteRecord = MessageFormatRecord.deserializeDeleteRecord(streamlog);
-          isDeleted = true;
-          deleteMsg = "delete change : AccountId:" + deleteRecord.getAccountId() + ", ContainerId:"
-              + deleteRecord.getContainerId() + ", DeletionTimeInSecs:" + deleteRecord.getDeletionTimeInMs();
+          encryptionKey = "No encryptionKey ";
```

Comment:
why this? Won't null do?

Reply:
Yes. But again while printing it will be null or I'd have to check for null and skip printing. So, went with this approach